walk/jujutsu_test.go

```diff
@@ -0,0 +1,87 @@
+package walk_test
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/numtide/treefmt/v2/stats"
+	"github.com/numtide/treefmt/v2/test"
+	"github.com/numtide/treefmt/v2/walk"
+	"github.com/stretchr/testify/require"
+)
+
+func TestJujutsuReader(t *testing.T) {
+	as := require.New(t)
+
+	tempDir := test.TempExamples(t)
+
+	// init a jujutsu repo (disable signing for testing)
+	cmd := exec.Command("jj", "git", "init", "--config", "signing.backend=none")
+	cmd.Dir = tempDir
+	as.NoError(cmd.Run(), "failed to init jujutsu repository")
+
+	// track new files (disable signing for testing), this step is required based on the jujutsu configuration.
+	// I prefer `snapshot.auto-track=none()`, which doesn't track new files automatically.
+	cmd = exec.Command("jj", "file", "track", "--config", "signing.backend=none", ".")
+	cmd.Dir = tempDir
+	as.NoError(cmd.Run(), "failed to add everything to the index")
+
+	// update jujutsu's index (disable signing for testing)
+	cmd = exec.Command("jj", "--config", "signing.backend=none")
+	cmd.Dir = tempDir
+	as.NoError(cmd.Run(), "failed to update the index")
+
+	// read empty worktree
+	statz := stats.New()
+	reader, err := walk.NewJujutsuReader(tempDir, "", &statz)
+	as.NoError(err)
+
+	files := make([]*walk.File, 33)
```

Comment:
nit: The number 33 looks pretty magical here. Is there some way to get rid of it? If not, could you add some explanation?